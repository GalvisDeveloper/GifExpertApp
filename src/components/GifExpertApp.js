

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import AddCategory from './AddCategory';
import { GifPanel } from './GifPanel';

const GifExpertApp = ({ title }) => {

    const [categories, setCategories] = useState(['Motogp']);

    return (
        <div className="container">
            <div className="title">
                <h2 className="animate__animated animate__bounce animate__slideInDown animate_fast">{title}</h2>
            </div>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map(category => (
                        <GifPanel
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>
        </div>
    );
}

GifExpertApp.prototype = {
    title: PropTypes.string.isRequired
}

export default GifExpertApp;
