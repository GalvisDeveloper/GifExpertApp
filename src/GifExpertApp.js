

import React, { useState } from 'react';
import PropTypes from 'prop-types';

import AddCategory from './components/AddCategory';
import { GifPanel } from './components/GifPanel';
import uniqid from 'uniqid';

const GifExpertApp = ({ mensaje }) => {

    const [categories, setCategories] = useState(['Primera', 'Segunda', 'Tercera']);

    return (
        <>
            <h2>{mensaje}</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                {
                    categories.map(category => {
                        return <GifPanel
                            category={category}
                            key={uniqid()}
                        />
                    })
                }
            </ol>

        </>
    );
}

GifExpertApp.prototype = {
    mensaje: PropTypes.string.isRequired
}


export default GifExpertApp;
