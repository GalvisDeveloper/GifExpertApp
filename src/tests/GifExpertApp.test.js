
import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../components/GifExpertApp';
import '@testing-library/jest-dom';

describe('Pruebas al componente <GifExpertApp />', () => {
    test('Prueba 1: El componente debe coincidir con el snapshot ', () => {
        const title = "Search any Gif that you want"
        const wrapper = shallow(<GifExpertApp title={title} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Prueba 2:  Debe mostrar una lista de categorias ', () => {
        const categories = ['Moto gp', 'Formula 1'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        // expect(wrapper.find('GifPanel').length).toBe(categories.length); *Enable comments in GifExpertApp component to do this one*
    });
});
