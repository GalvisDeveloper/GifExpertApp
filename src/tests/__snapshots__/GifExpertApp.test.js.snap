// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifExpertApp /> Prueba 1: El componente debe coincidir con el snapshot  1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    <h2
      className="animate__animated animate__bounce animate__slideInDown animate_fast"
    >
      Search any Gif that you want
    </h2>
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifPanel
      category="Moto gp"
      key="Moto gp"
    />
  </ol>
</div>
`;

exports[`Pruebas al componente <GifExpertApp /> Prueba 2:  Debe mostrar una lista de categorias  1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    <h2
      className="animate__animated animate__bounce animate__slideInDown animate_fast"
    />
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifPanel
      category="Moto gp"
      key="Moto gp"
    />
  </ol>
</div>
`;
